src/celeritas/field/CartMapFieldParams.hh

```diff
@@ -0,0 +1,82 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/field/CartMapFieldParams.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+
+#include "corecel/Config.hh"
+
+#include "corecel/Assert.hh"
+#include "corecel/data/ParamsDataInterface.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+struct CartMapFieldInput;
+class CartMapFieldParamsImpl;
```

Comment:
Instead of a `celeritas` namespace class, you should be able to make this a forward declared nested struct like the example I sent. (`struct CartMapFieldParams::Impl`). Then you can also use the uniqueptr deleter to avoid violating the rule of five.